From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nathan Froyd <froydnj@mozilla.com>
Date: Thu, 30 Jul 2020 15:55:30 +0000
Subject: [PATCH] Bug 1655929 - bump upstream versions of rlbox + lucet;
 r=firefox-build-system-reviewers,rstewart

We need this bump for a couple of reasons:

1. It enables floating the libc crate version, which makes keeping up-to-date
   with upstream changes easier.
2. It enables floating the getrandom crate version, which enables updating
   getrandom to a version with less buggy behavior around the Linux
   `getrandom` syscall on some architectures.

We also take this opportunity to sync up our vendored lucet with the lucet
that we use to build things in taskcluster.

Differential Revision: https://phabricator.services.mozilla.com/D85408
---
 .cargo/config.in                    | 4 ++--
 Cargo.toml                          | 2 +-
 taskcluster/ci/fetch/toolchains.yml | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/.cargo/config.in b/.cargo/config.in
index 37fac6a1008a83849771ee5f970df9ef776ae187..b5e11750783e00d274f99cd782e2c05191d34d99 100644
--- a/.cargo/config.in
+++ b/.cargo/config.in
@@ -70,12 +70,12 @@ rev = "64af847bc5fdcb6d2438bec8a6030812a80519a5"
 [source."https://github.com/PLSysSec/rlbox_lucet_sandbox/"]
 git = "https://github.com/PLSysSec/rlbox_lucet_sandbox/"
 replace-with = "vendored-sources"
-rev = "d510da5999a744c563b0acd18056069d1698273f"
+rev = "9bf4cdada11b2afd18efae5520bb8d3bb5bbc785"
 
 [source."https://github.com/PLSysSec/lucet_sandbox_compiler"]
 git = "https://github.com/PLSysSec/lucet_sandbox_compiler"
 replace-with = "vendored-sources"
-rev = "5e870faf6f95d79d11efc813e56370ad124bbed5"
+rev = "6594bb9dfab7e67eb6eb2fefbc66ab7c491ce730"
 
 [source."https://github.com/ChunMinChang/cubeb-coreaudio-rs"]
 git = "https://github.com/ChunMinChang/cubeb-coreaudio-rs"
diff --git a/Cargo.toml b/Cargo.toml
index 897daad41bed7b3ca6589a696b5cf7085556b438..568ea00e2f2949583ebca7d23b690cab5461407b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67,7 +67,7 @@ panic = "abort"
 [patch.crates-io]
 libudev-sys = { path = "dom/webauthn/libudev-sys" }
 packed_simd = { git = "https://github.com/hsivonen/packed_simd", rev="07d285ba65d9a4e3b9d76eacbf4c6489e125127d" }
-rlbox_lucet_sandbox = { git = "https://github.com/PLSysSec/rlbox_lucet_sandbox/", rev="d510da5999a744c563b0acd18056069d1698273f" }
+rlbox_lucet_sandbox = { git = "https://github.com/PLSysSec/rlbox_lucet_sandbox/", rev="9bf4cdada11b2afd18efae5520bb8d3bb5bbc785" }
 nix = { git = "https://github.com/shravanrn/nix/", rev="4af6c367603869a30fddb5ffb0aba2b9477ba92e" }
 spirv_cross = { git = "https://github.com/kvark/spirv_cross", branch = "wgpu3" }
 # failure's backtrace feature might break our builds, see bug 1608157.
diff --git a/taskcluster/ci/fetch/toolchains.yml b/taskcluster/ci/fetch/toolchains.yml
index 316fa9639edc0c4a4f606722003795e2f3d65f28..71c558516e71adfb32aff4f04e48f475841ddc08 100755
--- a/taskcluster/ci/fetch/toolchains.yml
+++ b/taskcluster/ci/fetch/toolchains.yml
@@ -463,7 +463,7 @@ lucetc-source:
     fetch:
         type: git
         repo: https://github.com/PLSysSec/lucet_sandbox_compiler/
-        revision: a0a1ce981638109f1f76cd1eecf17a0436a20055
+        revision: 6594bb9dfab7e67eb6eb2fefbc66ab7c491ce730
 
 dump-syms:
     description: dump_syms source code
